module/purpose/PS3Walk/src/PS3Walk.hpp

```diff
@@ -17,61 +17,76 @@
         static constexpr uint AXIS_ACCEL_Z                   = 25;
 
         // buttons
-        static constexpr uint BUTTON_SELECT         = 0;
-        static constexpr uint BUTTON_LEFT_JOYSTICK  = 1;
-        static constexpr uint BUTTON_RIGHT_JOYSTICK = 2;
-        static constexpr uint BUTTON_START          = 3;
-        static constexpr uint BUTTON_DPAD_UP        = 4;
-        static constexpr uint BUTTON_DPAD_RIGHT     = 5;
-        static constexpr uint BUTTON_DPAD_DOWN      = 6;
-        static constexpr uint BUTTON_DPAD_LEFT      = 7;
-        static constexpr uint BUTTON_L2             = 8;
-        static constexpr uint BUTTON_R2             = 9;
-        static constexpr uint BUTTON_L1             = 10;
-        static constexpr uint BUTTON_R1             = 11;
-        static constexpr uint BUTTON_TRIANGLE       = 12;
-        static constexpr uint BUTTON_CIRCLE         = 13;
-        static constexpr uint BUTTON_CROSS          = 14;
-        static constexpr uint BUTTON_SQUARE         = 15;
+        static constexpr uint BUTTON_SELECT         = 8;
+        static constexpr uint BUTTON_LEFT_JOYSTICK  = 11;
+        static constexpr uint BUTTON_RIGHT_JOYSTICK = 12;
+        static constexpr uint BUTTON_START          = 9;
+        static constexpr uint BUTTON_DPAD_UP        = 13;
+        static constexpr uint BUTTON_DPAD_RIGHT     = 16;
+        static constexpr uint BUTTON_DPAD_DOWN      = 14;
+        static constexpr uint BUTTON_DPAD_LEFT      = 15;
+        static constexpr uint BUTTON_L2             = 6;
+        static constexpr uint BUTTON_R2             = 7;
+        static constexpr uint BUTTON_L1             = 4;
+        static constexpr uint BUTTON_R1             = 5;
+        static constexpr uint BUTTON_TRIANGLE       = 2;
+        static constexpr uint BUTTON_CIRCLE         = 1;
+        static constexpr uint BUTTON_CROSS          = 0;
+        static constexpr uint BUTTON_SQUARE         = 3;
+        static constexpr uint BUTTON_HOME           = 10;
 
         /// @brief Called by the powerplant to build and setup the PS3Walk reactor.
         explicit PS3Walk(std::unique_ptr<NUClear::Environment> environment);
 
     private:
+        /// @brief Stores configuration values
+        struct Config {
+            float maximum_forward_velocity    = 0;
+            float maximum_rotational_velocity = 0;
+        } cfg;
+
+        /// @brief
         Joystick joystick;
-        Eigen::Vector2d strafe = Eigen::Vector2d::Zero();
-        float rotationalSpeed  = 0.0f;
-        bool moving            = false;
-        bool headLocked        = false;
-        float headPitch        = 0.0f;
-        float headYaw          = 0.0f;
+
+        /// @brief
+        Eigen::Vector3f walk_command = Eigen::Vector3f::Zero();
+
+        /// @brief
+        bool moving = false;
+
+        /// @brief
+        bool head_locked = false;
+
+        /// @brief
+        float head_pitch = 0.0f;
```

Comment:
Change the type of all the variables from `float` to `double`.  With Eigen types, `Eigen::Vector3f` for example would become `Eigen::Vector3d`